### PR TITLE
Refine notes implementation 

### PR DESCRIPTION
**Pull request by fabubaker**
- Add a validator for the notes column.
- Allow auto-migrations to run for notes routes by defining excluded routes more strictly.
- Fix notes actions being added to customizations log with an incorrect key.
- Separate legacy and rayyan notes when importing/exporting.

### CODE REVIEW

File: server/src/app.ts [hunk 1]
```diff
@@ -47,10 +47,20 @@ app.use(function (req, res, next) {
...
+    '/reviews/[0-9]+/version$', // getReviewVersion
+    '/reviews/[0-9]+/downgrade$' // downgradeReview
+  ]
+
+  // If req.path matches any of the above regexes,
+  // skip this middleware.
+  for (const regexStr of excludeRegexes) {
+    const regex = new RegExp(regexStr)
+    if (regex.test(req.path)) {
+      return next()
```
Comment (hammady) [inline server/src/app.ts:62]: Being in a for loop, don't we want to break once the regex matches?

File: server/src/app.ts [hunk 2]
```diff
@@ -47,10 +47,20 @@ app.use(function (req, res, next) {
   // - deleteReview
   // - getReviewVersion
   // - downgradeReview
-  const re = new RegExp('/reviews/.*');
 
-  if (re.test(req.path)) {
-    return next()
+  const excludeRegexes = [
```
Comment (hammady) [inline server/src/app.ts:51]: What about being more explicit and match the http verbs as well?

File: server/src/adapters/postgres/postgres.ts [hunk 3]
```diff
@@ -447,7 +465,7 @@ const reviewColumns = [
   { 'colName': 'dedup_results', 'validator': validateDedupResult, 'defaultVal': {} },
   { 'colName': 'suggestion_ratings', 'validator': validateSuggestionRating, 'defaultVal': {} },
   { 'colName': 'fulltexts', 'validator': validateFulltextColumn, 'defaultVal': { 'public': {}, 'private': {} } },
-  { 'colName': 'notes',  'defaultVal': {} },
+  { 'colName': 'notes',  'validator': validateNotesColumn, 'defaultVal': {}, 'keyName': 'rayyan_notes' },
```
Comment (hammady) [inline server/src/adapters/postgres/postgres.ts:468]: I thought we are calling it just `notes`?

**Review by hammady — CHANGES_REQUESTED** [review]

File: server/src/app.ts (hunk 1 above)
Comment (fabubaker) [reply server/src/app.ts:62]: If the regex matches, the auto-migration part (which is below) is skipped as the middleware returns early.

File: server/src/adapters/postgres/postgres.ts (hunk 3 above)
Comment (fabubaker) [reply server/src/adapters/postgres/postgres.ts:468]: The column in the db is still called notes, but when you import/export, the key in the JSON is expected to be `rayyan_notes`.

This is to separate legacy and rayyan notes when importing/exporting.

**Review by hammady — APPROVED** [review]

**Pull request merged**